src/docs/asciidoc/reference/architecture/code-structure.adoc

```diff
@@ -1,2 +1,150 @@
+
+By default (When the application is generated by the _CLI_), the `worker` folder contains only one file, called `index.ts`.
+
+include::{docdir}/common/code/generated-worker-code.adoc[]
+
+.Develop your worker in several ways
+[NOTE]
+====
+In this example we see a worker that is created with only one class. You will see in the <<reference.adoc#develop_custom_cloud_services, cloud service development section>> how to create a worker in several ways.
+====
+
+.Don't need to transpile TypeScript
+[NOTE]
+====
+Even if your worker code is in `TypeScript` you don't need to transpile it to `JavaScript`. Indeed, when you run or push your code, it will be launched with link:{ts-node-url}[`ts-node`]. +
+`ts-node` take care of the transpilation, you can also configure this via the `tsconfig.json` file.
+====
+
+=== front folder
+
+The `front` folder contains your front code.
+So you need to have your `index.html` (or other html file if you want) at the root of this folder.
+
+.Use frameworks
+[NOTE]
+====
+You can integrate your favorite framework with ZetaPush.
+For now, we give you documentation for integration with the most famous frameworks : <<reference#develop_with_angular, Angular>>,
+<<reference#develop_with_react, React>> and <<reference#develop_with_vue, Vue.js>>.
+====
+
+=== application.json
+
+This file is *optional* in a *ZetaPush application*. For now you just need to use it when you use the <<developer-manual.adoc#user-management-in-application, StandardUserWorkflow>>.
+Of course, you can also use it without using the <<developer-manual.adoc#user-management-in-application, StandardUserWorkflow>>.
+
+include::{docdir}/common/code/application-json-standard-user-workflow.adoc[]
+
+.List of application.json properties
+[NOTE]
+====
+To know which keys are available for the `application.json` file, please refer you to the _cloud service_ that use this configuration file like the <<developer-manual.adoc#user-management-in-application, StandardUserWorkflow>>.
+You can also <<reference#configuration_properties, define your own properties>>.
+====
+
+=== .zetarc
+
+The `.zetarc` file is the most important file in your application.
+It lets you define your *ZetaPush account* and which *ZetaPush application* you want to use.
+
+include::{docdir}/common/code/minimal-zetarc.adoc[]
+
+In addition to the properties displayed in the example, 2 properties are used:
+
+* *_developerSecretToken_*: The _CLI_ encrypts the `developerPassword` in the `.zetarc` file. You don't need to modify it. If you want to change the ZetaPush account password in the `.zetarc` file, you can delete the `developerSecretToken` and type again the `developerPassword`.
+* *_platformUrl_*: If you don't know what does it matter, don't worry about that, by default you don't need to set it. That corresponds to the used *ZetaPush Cloud* for your *ZetaPush application*. If you need it, our CSO provided it to you.
```

Comment:
If you don't know what does it matter, don't worry about that, by default you don't need to set it. That corresponds to the used *ZetaPush Cloud* for your *ZetaPush application*. If you need it, our CSO provided it to you. -> That corresponds to the used *ZetaPush Cloud* for your *ZetaPush application*. When using ZetaPush public cloud, you don't need to configure it. If you need a private ZetaPush cloud, you can link:{contact-url}[contact us] to ask for it.